### PR TITLE
chore: v4.0.1 - update changelog and bump cargo versions 

### PR DESCRIPTION
**Pull request by jgreat**
### Motivation

Update changelog with and bump cargo versions to prepare for the v4.0.1 release.

### CODE REVIEW

**Review by jcape — APPROVED** [review]
This doesn't update the changelog, fwiw

**Comment by jcape** [comment]
Also needs lock files to be updated

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,9 +6,13 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
...
 
-## Unreleased
+## [4.0.1]
 
-- Improved performance of mobilecoind-dev-faucet for better load testing
+### Added/Changed
+
+- Improved performance of mobilecoind-dev-faucet for better load testing ([#2955])
+- Make the local network script support a dense3 network option ([#2988])
+- HAProxy Ingress Blocklists ([#2952])
```
Comment (nick-mobilecoin) [inline CHANGELOG.md:15]: :upside_down_face: Suggest separate sections for "Added" and "Changed" 
```suggestion
## [4.0.1]

### Added

- Make the local network script support a dense3 network option ([#2988])
- HAProxy Ingress Blocklists ([#2952])

### Changed

- Improved performance of mobilecoind-dev-faucet for better load testing ([#2955])
```

**Review by nick-mobilecoin — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (jcape) [reply CHANGELOG.md:15]: This should get applied.

**Pull request merged**